types/tizen-common-web/lib/tvinfo.d.ts

```diff
@@ -0,0 +1,305 @@
+    | 'CAPTION_FONT_STYLE1'
+    | 'CAPTION_FONT_STYLE2'
+    | 'CAPTION_FONT_STYLE3'
+    | 'CAPTION_FONT_STYLE4'
+    | 'CAPTION_FONT_STYLE5'
+    | 'CAPTION_FONT_STYLE6'
+    | 'CAPTION_FONT_STYLE7';
+
+/**
+ * Available values for the caption menu color.
+ * These values may be returned for keys CAPTION_FONT_COLOR_KEY, CAPTION_BG_COLOR_KEY, CAPTION_EDGE_COLOR_KEY and CAPTION_WINDOW_COLOR_KEY.
+ * @since 2.4
+ */
+declare const enum CaptionColor {
+    CAPTION_COLOR_DEFAULT = 'CAPTION_COLOR_DEFAULT',
+    CAPTION_COLOR_WHITE = 'CAPTION_COLOR_WHITE',
+    CAPTION_COLOR_BLACK = 'CAPTION_COLOR_BLACK',
+    CAPTION_COLOR_RED = 'CAPTION_COLOR_RED',
+    CAPTION_COLOR_GREEN = 'CAPTION_COLOR_GREEN',
+    CAPTION_COLOR_BLUE = 'CAPTION_COLOR_BLUE',
+    CAPTION_COLOR_YELLOW = 'CAPTION_COLOR_YELLOW',
+    CAPTION_COLOR_MAGENTA = 'CAPTION_COLOR_MAGENTA',
+    CAPTION_COLOR_CYAN = 'CAPTION_COLOR_CYAN',
+}
+
+type CaptionColorUnion =
+    | CaptionColor
+    | 'CAPTION_COLOR_DEFAULT'
+    | 'CAPTION_COLOR_WHITE'
+    | 'CAPTION_COLOR_BLACK'
+    | 'CAPTION_COLOR_RED'
+    | 'CAPTION_COLOR_GREEN'
+    | 'CAPTION_COLOR_BLUE'
+    | 'CAPTION_COLOR_YELLOW'
+    | 'CAPTION_COLOR_MAGENTA'
+    | 'CAPTION_COLOR_CYAN';
+
+/**
+ * Available values for the caption menu opacity.
+ * These values may be returned for keys CAPTION_FONT_OPACITY_KEY, CAPTION_BG_OPACITY_KEY and CAPTION_WINDOW_OPACITY_KEY.
+ * @since 2.4
+ */
+declare const enum CaptionOpacity {
+    CAPTION_OPACITY_SOLID = 'CAPTION_OPACITY_SOLID',
+    CAPTION_OPACITY_FLASHING = 'CAPTION_OPACITY_FLASHING',
+    CAPTION_OPACITY_TRANSLUCENT = 'CAPTION_OPACITY_TRANSLUCENT',
+    CAPTION_OPACITY_TRANSPARENT = 'CAPTION_OPACITY_TRANSPARENT',
+    CAPTION_OPACITY_DEFAULT = 'CAPTION_OPACITY_DEFAULT',
+}
+
+type CaptionOpacityUnion =
+    | CaptionOpacity
+    | 'CAPTION_OPACITY_SOLID'
+    | 'CAPTION_OPACITY_FLASHING'
+    | 'CAPTION_OPACITY_TRANSLUCENT'
+    | 'CAPTION_OPACITY_TRANSPARENT'
+    | 'CAPTION_OPACITY_DEFAULT';
+
+/**
+ * Available values for the caption menu edge type.
+ * These values may be returned for key CAPTION_EDGE_TYPE_KEY.
+ * - `CAPTION_EDGE_NONE` - no edge
+ * - `CAPTION_EDGE_RAISED` - raised edge
+ * - `CAPTION_EDGE_DEPRESSED` - depressed edge
+ * - `CAPTION_EDGE_UNIFORM` - uniform edge
+ * - `CAPTION_EDGE_DROP_SHADOWED` - drop shadowed edge
+ * @since 2.4
+ */
+declare const enum CaptionEdge {
+    CAPTION_EDGE_NONE = 'CAPTION_EDGE_NONE',
+    CAPTION_EDGE_RAISED = 'CAPTION_EDGE_RAISED',
+    CAPTION_EDGE_DEPRESSED = 'CAPTION_EDGE_DEPRESSED',
+    CAPTION_EDGE_UNIFORM = 'CAPTION_EDGE_UNIFORM',
+    CAPTION_EDGE_DROP_SHADOWED = 'CAPTION_EDGE_DROP_SHADOWED',
+}
+
+type CaptionEdgeUnion =
+    | CaptionEdge
+    | 'CAPTION_EDGE_NONE'
+    | 'CAPTION_EDGE_RAISED'
+    | 'CAPTION_EDGE_DEPRESSED'
+    | 'CAPTION_EDGE_UNIFORM'
+    | 'CAPTION_EDGE_DROP_SHADOWED';
+/**
+ * All available values for the caption menu.
+ */
+declare namespace Tizen {
+    type CaptionValue = CaptionStateUnion &
```

Comment:
This is an intersection of unions of literal types, and is equivalent to `never`, if I'm not mistaken. I don't think that's the intent?

Reply:
Before discuss about it, I modify the `CaptionStateUnion` to `CaptionUnion`. We want to support to developer about both a `string` type and a enumeration. But the `xxxxUnion` make it misunderstood, so we removed the `xxxxUnion`.

Now let's talk about the intersection.
```js
interface TVInfoManager {
        /**
         * Method returns the value for corresponding caption menu key.
         * @returns CaptionValue value for given caption menu key
         * @param key Caption menu
         * @throw WebAPIException TypeMismatchError, UnknownError
         * @since 2.4
         */
        getCaptionValue(key: CaptionInfoKeyUnion): CaptionValue;
```

Refer to the `getCaptionValue`, this method return a string value of the enum which is `CaptionStateUnion`, `    CaptionModeUnion`, `CaptionFontSizeUnion`, `CaptionFontStyleUnion`, `CaptionColorUnion`, `CaptionOpacityUnion` or `     CaptionEdgeUnion`. Thus, I think using intersection of enum is better than enumerate `string`s.

Reply:
You'd want a union of the enums, then, no? Since it can return a value from any one of the enums?

Reply:
The `getCaptionValue` return any one of the enums. 
And I got mistake, the `getCaptionValue` returns a intersection of enums like 
``` ts
type CaptionValue = CaptionState &
        CaptionMode &
        CaptionFontSize &
        CaptionFontStyle &
        CaptionColor &
        CaptionOpacity &
        CaptionEdge;
```